### PR TITLE
Update VThread alive check to check continuation state

### PR DESCRIPTION
**Pull request by fengxue-IS**
Related: https://github.com/eclipse-openj9/openj9/issues/18738#issuecomment-2218824062

### CODE REVIEW

**Comment by fengxue-IS** [comment]
Local testing  passed 10/10 on failed test case.
Note this test have been [disabled due to inconsistent result by openjdk](https://bugs.openjdk.org/browse/JDK-8316397)

**Comment by babsingh** [comment]
We should add a more detailed commit message.

```
Update virtual thread alive check in JVMTI getVMThread

Check if the continuation is finished to determine more accurately
if a virtual thread is alive.

Related: https://github.com/eclipse-openj9/openj9/issues/18738#issuecomment-2218824062
```

File: runtime/jvmti/jvmtiHelpers.cpp [hunk 1]
```diff
@@ -178,7 +178,9 @@ getVMThread(J9VMThread *currentThread, jthread thread, J9VMThread **vmThreadPtr,
 				targetThread = carrierVMThread;
 			}
 		}
-		isThreadAlive = (JVMTI_VTHREAD_STATE_NEW != vthreadState) && (JVMTI_VTHREAD_STATE_TERMINATED != vthreadState);
+		j9object_t continuationObj = J9VMJAVALANGVIRTUALTHREAD_CONT(currentThread, threadObject);
```
[the remark this answers is not in the document]
Comment (fengxue-IS) [reply runtime/jvmti/jvmtiHelpers.cpp:181]: yes, continuation object is filled as part of the constructor, so any vthread object that can be passed into JVMTI should have a continuation object associated.

File: runtime/jvmti/jvmtiHelpers.cpp [hunk 2]
```diff
@@ -178,7 +178,9 @@ getVMThread(J9VMThread *currentThread, jthread thread, J9VMThread **vmThreadPtr,
...
+		ContinuationState continuationState = *VM_ContinuationHelpers::getContinuationStateAddress(currentThread, continuationObj);
+		isThreadAlive = (JVMTI_VTHREAD_STATE_NEW != vthreadState) && (JVMTI_VTHREAD_STATE_TERMINATED != vthreadState) && !VM_ContinuationHelpers::isFinished(continuationState);
```
[the remark this answers is not in the document]
Comment (fengxue-IS) [reply runtime/jvmti/jvmtiHelpers.cpp]: fixed

**Comment by babsingh** [comment]
Jenkins test sanity,sanity.openjdk,extended.openjdk amac jdk23

**Comment by babsingh** [comment]
Only known/unrelated failures are seen.

image

**Review by babsingh — APPROVED** [review]

**Comment by babsingh** [comment]
@fengxue-IS Can you backport the fix to the [v0.47.0-release](https://github.com/thallium/openj9/tree/v0.47.0-release) branch?

**Pull request merged**